### ISSUE TITLE
cannot login in some cases

### ISSUE THREAD

**Issue by ifuratdev** [luyadev/luya-module-admin#345]
### What steps will reproduce the problem?
i'm facing a strange behavior
I can login from the desktop
I can login from mobile android chrome incognito mode
But I cannot login from mobile android chrome
It keeps redirecting me to the login page.

I have debugged the code and the login is successful, but it keeps redirecting me to the login page

 

### Additional infos

| Q                | A
| ---------------- | ---
| LUYA Version     | 
| PHP Version      |
| Platform         | Apache/XAMPP/MAMPP/etc.
| Operating system | Windows/Linux Server/OSX/etc.

**Comment by nadar** [comment]
try to disable debug toolbar. otherwise its a problem of your server or device

**Comment by nadar** [comment]
and also make not sure you are already logged in with this user in another window. a user can only logged in once.

**Comment by ifuratdev** [comment]
yii toolbar already disabled
I have tried with two different phones and I faced the exact same behavior

It works fine using chrome incognito mode
but it keeps redirecting to the login page after successful login

**Comment by ifuratdev** [comment]
i've checked
https://github.com/luyadev/luya-module-admin/issues/45

and FileCache is used in the config
yii toolbar is already disabled

**Comment by ifuratdev** [comment]
I want to test changing the return URL after successful login

Where to configure this url in Luya?

**Comment by ifuratdev** [comment]
Initial investigation:
Android Chrome browser has a functionality of loading the data from Google Proxy server.

This causes Yii::$app->request->userIP  to return Google proxy server instead of the user ip address

So in when requesting the IP address of the user using "$this->ip = Yii::$app->request->userIP"
in
vendor\luyadev\luya-module-admin\src\models\UserLogin.php
and 
vendor\luyadev\luya-module-admin\src\models\User.php
and
vendor/luyadev/luya-module-admin/src/controllers/DefaultController.php

The wrong ip address (Google Proxy Sever) is returned and the login is failed

i'm investigating using $_SERVER['HTTP_FORWARDED'] instead of "$this->ip = Yii::$app->request->userIP"

**Comment by ifuratdev** [comment]
yes, the above is the solution, it works now

Full day of investigation to resolve this issue

Hope this helps any one facing the same issue

**Comment by ifuratdev** [comment]
temporary code using to fix the issue in all php files 

vendor\luyadev\luya-module-admin\src\models\UserLogin.php
and
vendor\luyadev\luya-module-admin\src\models\User.php
and
vendor/luyadev/luya-module-admin/src/controllers/DefaultController.php


		$temp_ip_address = 0;
		if(isset($_SERVER['HTTP_FORWARDED'])){
			$temp_ip_address = substr($_SERVER['HTTP_FORWARDED'], 4);
		}else{
			$temp_ip_address = Yii::$app->request->userIP;
		}

**Comment by nadar** [comment]
Thanks for the research. Using HTTP_FORWARDED can be a risk i assume, as it could be spoofed, but i am not sure. So the problem is that you change the ip on every request? Maybe we could add an option to disable this feature, but in terms of security it's a good behavior.

**Comment by ifuratdev** [comment]
the change in IP address is done by the browser for faster browsing.
So yes, I think the reason is the change in IP address

**Comment by nadar** [comment]
So we can close this issue with the research, in favor of this https://github.com/luyadev/luya-module-admin/issues/346

**Comment by ifuratdev** [comment]
yes,
I have tested the code and it is due to https://github.com/luyadev/luya-module-admin/issues/346

**Comment by svetlika** [comment]
Hi! I have the same problem.

I just installed luya-kickstarter and I cannot login into the admin area. After successeful login it shows me big green circle with chect and than redirects me again to the page admin/login.
What should I do?

I tried to replace "Yii::$app->request->userIP" with code that suggested ifuratdev, but nothing changed. It still redirects me to the admin/login.

**Comment by nadar** [comment]
hi @svetlika try to disabled `$logoutOnUserIpChange` in admin module config settings:

https://github.com/luyadev/luya-module-admin/blob/master/src/Module.php#L264-L270

**Comment by svetlika** [comment]
hi @nadar I can't disable it, because it is doesn't exist in my installation of Luya.
Anyway, I pasted the highlighted code into my vendor/luyadev/luya-module-admin/src/Module.php, and I changed "true" to "false", but as I thought, nothing changed. I still cannot login into Luya after installation.

Sorry, if I misplaced my comment, It is my first experience with Luya.
I couldn't find here https://luya.io/guide/installation/ anything about this problem and how to solve it. :(

**Comment by nadar** [comment]
Of course you can :-) Maybe post your config, so we can help. But it should look like this:

```
 'modules' => [
        // See all options: https://luya.io/api/luya-admin-Module
        'admin' => [
            'class' => 'luya\admin\Module',
           'logoutOnUserIpChange' => false, // <-------------------------------------------------------------
            'secureLogin' => false,
            'strongPasswordPolicy' => false, 
            'interfaceLanguage' => 'en',
            'autoBootstrapQueue' => true,
        ],
        'cms' => [
            'class' => 'luya\cms\frontend\Module',
            'contentCompression' => false,
        ],
        'cmsadmin' => [
            'class' => 'luya\cms\admin\Module',
        ],
    ],
```

**Comment by nadar** [comment]
https://luya.io/guide/installation/problems.html#logout-after-login

**Comment by svetlika** [comment]
Oh, sorry. I really didn't see this section on the page with documentation. :(

But as I noticed (because of the big difference between your and mine versions of "luya-module-admin/Module.php"), the version of the luya-kickstarter which I installed yesterday was very-very old. 
So, I updated it, and I can login now.

Maybe you should edit the Guide (https://luya.io/guide/installation/) and modify there the version of the composer command of installing? Or mention somewhere which version is now actual at least. It is not so obvious to a completely newbie like me.

Anyway, thank you for your help!

**Comment by nadar** [comment]
Can you show me how you get to the old version? Because that kickstarters requires the latest deps => https://github.com/luyadev/luya-kickstarter/blob/master/composer.json

Would be nice to know, thanks

**Comment by svetlika** [comment]
I run this command in my console:

> composer create-project luyadev/luya-kickstarter:^1.0

I copied it from your guide here: https://luya.io/guide/installation/#create-project

I thought it would be the last version of Luya, but it was quite old as I understood this morning.
This was in composer.json which I received by that command:

{
    "name": "luyadev/luya-kickstarter",
    "description": "The kickstarter project for a new LUYA application.",
    "type": "project",
    "license":"MIT",
    "require": {
        "luyadev/luya-core" : "~1.0.0",
        "luyadev/luya-module-admin" : "~1.0.0",
        "luyadev/luya-module-cms" : "~1.0.0",
        "luyadev/luya-bootstrap3" : "~1.0.0",
        "luyadev/luya-generic" : "~1.0.0"
    },
    "require-dev": {
        "yiisoft/yii2-gii" : "~2.0.0",
        "yiisoft/yii2-debug" : "~2.0.0",
        "luyadev/luya-testsuite" : "~1.0.0"
    },
....

I couldn't find the number of actual version of Luya-kickstarter or Luya on the website or in this section of guide, so I assumed yesterday that it was the last, but, alas...

**Comment by nadar** [comment]
If i run

> `composer create-project luyadev/luya-kickstarter:^1.0`

i get

```json
{
    "name": "luyadev/luya-kickstarter",
    "description": "The kickstarter project for a new LUYA application.",
    "type": "luya-kickstarter",
    "license":"MIT",
    "require": {
        "luyadev/luya-module-admin" : "^4.0",
        "luyadev/luya-module-cms" : "^4.0",
        "luyadev/luya-bootstrap3" : "^1.0",
        "luyadev/luya-generic" : "^1.0"
    },
    "require-dev": {
        "yiisoft/yii2-gii" : "^2.0",
        "yiisoft/yii2-debug" : "^2.0",
        "luyadev/luya-testsuite" : "^2.0"
    }
}
```

Which composer version do you have `composer -v`, i have tested on ` 2.4.1`

**Comment by svetlika** [comment]
It was 1.10.1.
I don't use it often though.
When I was trying to update my installation of Luya, I had to update composer too because of error with github token.

**Comment by svetlika** [comment]
Hi! As I wrote, after I updated luya-kickstarter the admin panel is works and now I can login. But the start page of app doesn't work. There are a bunch of Yii2 errors. 
I didn't do anything in admin, just have logined.

I assumed that something is not updated correctly, so I tried to install luya-kickstarter again from scratch with composer 2.5.1.

And again after the command "composer create-project luyadev/luya-kickstarter:^1.0" the composer.json file in luya-kickstarter folder contained this lines:

{
"name": "luyadev/luya-kickstarter",
"description": "The kickstarter project for a new LUYA application.",
"type": "project",
"license":"MIT",
"require": {
"luyadev/luya-core" : "~1.0.0",
"luyadev/luya-module-admin" : "~1.0.0",
"luyadev/luya-module-cms" : "~1.0.0",
"luyadev/luya-bootstrap3" : "~1.0.0",
"luyadev/luya-generic" : "~1.0.0"
},
"require-dev": {
"yiisoft/yii2-gii" : "~2.0.0",
"yiisoft/yii2-debug" : "~2.0.0",
"luyadev/luya-testsuite" : "~1.0.0"
},

This time the composer also throw an error during the execution of this command because of absence of fxp. 
Also there was no folder "vendor" in this installation.

I replace again composer.json with one from repo luya-kickstarter GitHub, update it by "composer update" and made other steps from installation guide. Unfortunately something got very very wrong. Now I cannot access even admin panel because of many Yii2 errors. The start page of app also doesn't work, but the set of Yii2 errors is different from my first attemption of luya-kickstarter's installation.

Sorry to bother you, I don't have much experience with composer installation and don't understand what is happened.
